docs/auth_server.rst

```diff
@@ -0,0 +1,179 @@
+actions. This is done using the ``--scope`` argument when registering the client. The
+possible values are:
+
+=============  ======================================================
+permission     explanation
+=============  ======================================================
+ARCHIVE_READ   The client can read items from archive collection
+DESK_READ      The client can get desks metadata
+PLANNING_READ  The client can access planning, events and assignments
+CONTACT_READ   The client can get contacts metadata
+USERS_READ     The client can retrieve users
+=============  ======================================================
+
+Access token
+------------
+
+If client is allowed, an access token is generated and can be used to access the desired
+resources. The token is a `JSON Web Token`_ (JWT), which is signed, allowing the resource
+server to verify it.
+
+The payload of the token will contain the following keys:
+
+=========  ====================================================================
+key        explanation
+=========  ====================================================================
+client_id  id of the allowed client
+iss        principal that issued the JWT, it's always ``Superdesk Auth Server``
+iat        time when the JWT was issued (Unix time)
+exp        expiration time (Unix time)
+scope      list of allowed scopes (see above)
+=========  ====================================================================
+
+.. _JSON Web Token: https://tools.ietf.org/html/rfc7519
+
+Security
+--------
+
+**All the traffic must be encrypted using HTTPS**.
+
+The initial request is done by the client using `HTTP Basic Authentication`_, meaning the
+password is going on the wire.
+
+A salted hash of the client *password* is stored in superdesk, along with *client id* and
+*scope*.
+
+The JWT access token is not stored, it is only validated by resource server by checking its
+signature.
+
+The authorisation server and the resource server share a secret to sign and validate the
+JWT Token. We recommend to use an environment variable instead of `settings.py` to set
+this secret (the name of the variable is ``AUTH_SERVER_SHARED_SECRET``).
+
+.. _HTTP Basic Authentication: https://tools.ietf.org/html/rfc7617
+
+Testing
+-------
+
+By default, unsecured HTTP requests will be rejected. If you want to test authorisation
+server with a local instance without HTTPS, you may set the ``AUTHLIB_INSECURE_TRANSPORT``
+environment variable in the shell where server is started::
+
+  $ export AUTHLIB_INSECURE_TRANSPORT=1
+
+This is only for testing/development, **do NOT do that in production**.
+
+To test locally a client token request, you can use curl:
+
+.. code:: sh
+
+    #!/bin/sh
+
+    CLIENT_ID=123456
+    CLIENT_SECRET=789101112
+    URL=http://127.0.0.1:5000/api/auth_server/token
+
+    curl -u ${CLIENT_ID}:${CLIENT_SECRET} -XPOST ${URL} -F grant_type=client_credential
```

Comment:
Should it be grant_type=client_credential**s**?

Reply:
indeed, good catch.